docs/usage/jssengine.md

```diff
@@ -0,0 +1,521 @@
+`unwrap(...)`) isn't supported.
+
+Checking the current mode can be done via `getUseClientMode(...)`.
+
+#### Choosing Key Material
+
+Key material can be chosen in several ways. In every scenario, a JSSKeyManager
+instance needs to be passed to the JSSEngine:
+
+```java
+// JSSEngine inst;
+inst.setKeyManager(new JSSKeyManager());
+```
+
+For direct selection of key from an existing instance, call `setKeyMaterials`:
+
+```java
+// JSSEngine inst;
+inst.setKeyMaterials(myPK11Cert, myPK11PrivKey);
+```
+
+Note that these must be instances of [`PK11Cert`][jss.pk11-cert] and
+[`PK11PrivKey`][jss.pk11-privkey] respectively. These can be obtained from
+the [`CryptoManager`][jss.cryptomanager] and casting `X509Certificate` to
+`PK11Cert` and `PrivateKey` to `PK11PrivKey`.
+
+For selection of a key via an alias in the certificate database, call
+`setCertFromAlias`:
+
+```java
+// JSSEngine inst;
+inst.setCertFromAlias("server-cert-alias");
+```
+
+Lastly, key material could've been provided when the `JSSEngine` was
+constructed; see the section on direct utilization above.
+
+Note that SNI support isn't yet added so the key selection must occur prior
+to the initial handshake.
+
+#### Choosing TLS protocol version
+
+There are two ways to choose TLS protocol version. The first is via the Java
+Provider interface, selecting the TLS version directly. The `Mozilla-JSS`
+provider understands the following aliases:
+
+ - `SSL` and `TLS`, enabling any allowed SSL and TLS protocol version,
+ - `TLSv1.1` to enable only TLS version 1.1 by default,
+ - `TLSv1.2` to enable only TLS version 1.2 by default, and
+ - `TLSv1.3` to enable only TLS version 1.3 by default.
+
+Alternatively, the standard `SSLEngine` configuration method of passing
+a list of protocols to `setEnabledProtocols` is also allowed. Note that this
+will override any value passed via the Provider interface. Additionally, due
+to restrictions in NSS, a contiguous range of protocols will be enabled. For
+example, the following call:
+
+```java
+// SSLEngine inst;
+inst.setEnabledProtocols(new String[] { "TLSv1.1", "TLSv1.3" });
+```
+
+will enable TLSv1.1, TLSv1.2, and TLSv1.3.
+
+Alternative methods are available that take JSS standard `SSLVersion` and
+`SSLVersionRange` values as parameters; see the `JSSEngine` javadoc for
+more information.
+
+#### Choosing Cipher Suite
+
+Configuring cipher suites is performed using the standard `SSLEngine`
+configuration method of passing a list of cipher suites to
+`setEnabledCipherSuites`. We filter the list of passed cipher suites to
+only those allowed by local policy. For example:
+
+```java
+// SSLEngine inst;
+inst.setEnabledCipherSuites(new String[] { "TLS_AES_128_GCM_SHA256" });
+```
+
+will enable only a single TLSv1.3 cipher suite.
+
+Alternative methods are available that take JSS standard `SSLCipher`
+values as parameters; see the `JSSEngine` javadoc for more information.
+
+#### Using `JSSParameters`
```

Comment:
This section might be a little confusing to those initiated. Here is just says it's mostly like SSLParameters but with added properties. Would one know how and when to use these instead of the SSLParameters? Also not those are mentioned indirectly when talking about  "SSLContext.getDefaultSSLParameters() and SSLContext.getSupportedSSLParameters()".  Only asking...

Reply:
I just noticed that there is a reference to the JSSParameters separate man page.

Reply:
OK clarified. `JSSParameters` inherits from `SSLParameters`; the latter is in the standard library and should be used by most everyone, but if you need something only `JSSParameters` provides (or wish to re-use existing JSS classes such as the `SSLCipher` enums) it is better to use `JSSParameters`.